scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/CheckpointSimulationFacade.java

```diff
@@ -0,0 +1,297 @@
+  @Override
+  public SimulationResultsComputerInputs simulateNoResultsUntilEndPlan(final Plan plan)
+  throws SimulationException, SchedulingInterruptedException
+  {
+    return simulateNoResults(plan, planningHorizon.getEndAerie(), null).simulationResultsComputerInputs();
+  }
+
+    /**
+   * Simulates a plan until the end of one of its activities
+   * Do not use to update the plan as decomposing activities may not finish
+   * @param plan
+   * @param activity
+   * @return
+   * @throws SimulationException
+   */
+
+  @Override
+  public SimulationResultsComputerInputs simulateNoResultsUntilEndAct(
+          final Plan plan,
+          final SchedulingActivityDirective activity) throws SimulationException, SchedulingInterruptedException
+  {
+    return simulateNoResults(plan, null, activity).simulationResultsComputerInputs();
+  }
+
+    public AugmentedSimulationResultsComputerInputs simulateNoResults(
+            final Plan plan,
+            final Duration until) throws SimulationException, SchedulingInterruptedException
+  {
+    return simulateNoResults(plan, until, null);
+  }
+
+
+    /**
+     * Simulates and updates plan
+     * @param plan
+     * @param until can be null
+     * @param activity can be null
+     */
+  private AugmentedSimulationResultsComputerInputs simulateNoResults(
+      final Plan plan,
+      final Duration until,
+      final SchedulingActivityDirective activity) throws SimulationException, SchedulingInterruptedException
+  {
+    final var planSimCorrespondence = scheduleFromPlan(plan, this.schedulerModel);
+
+    final var best = CheckpointSimulationDriver.bestCachedEngine(
+        planSimCorrespondence.directiveIdActivityDirectiveMap(),
+        cachedEngines.getCachedEngines(configuration));
+    CheckpointSimulationDriver.CachedSimulationEngine engine = null;
+    Duration from = Duration.ZERO;
+    if(best.isPresent()){
+      engine = best.get().getKey();
+      replaceIds(planSimCorrespondence, best.get().getRight());
+      from = engine.endsAt();
+    }
+
+    //Configuration
+    //Three modes : (1) until a specific end time (2) until end of one specific activity (3) until end of last activity in plan
+    Duration simulationDuration;
+    Function<CheckpointSimulationDriver.SimulationState, Boolean>
+        stoppingCondition;
+    //(1)
+    if(until != null && activity == null){
+      simulationDuration = until;
+      stoppingCondition = CheckpointSimulationDriver.noCondition();
+      LOGGER.info("Simulation mode: until specific time " + simulationDuration);
+    }
+    //(2)
+    else if(activity != null && until == null){
+      simulationDuration = planningHorizon.getEndAerie();
+      stoppingCondition = CheckpointSimulationDriver.stopOnceActivityHasFinished(planSimCorrespondence.planActDirectiveIdToSimulationActivityDirectiveId().get(activity));
+      LOGGER.info("Simulation mode: until activity ends " + activity);
+      //(3)
+    } else if(activity == null && until == null){
```

Comment:
Nitpick, nonblocking: formatting
```suggestion
    //(3)
    } else if(activity == null && until == null){
```